PetApiTest/PetAPITest.cs

```diff
@@ -0,0 +1,186 @@
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("petStore/removePets");
+            Pet pet = new Pet("huanhuan", "dog", "white", 5000);
+            string request = JsonConvert.SerializeObject(pet);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+            //when
+            await client.PostAsync("petStore/addNewPet", requestBody);
+            var response = await client.GetAsync("petStore/pets");
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actualPets = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Assert.Contains(pet, actualPets);
+        }
+
+        [Fact]
+        public async void Should_get_right_pet_When_search_by_name()
+        {
+            //given
+            TestServer server = new TestServer(new WebHostBuilder().UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("petStore/removePets");
+            Pet pet1 = new Pet("HuanHuan", "dog", "white", 5000);
+            Pet pet2 = new Pet("DaHuang", "dog", "white", 5000);
+            Pet pet3 = new Pet("WangCai", "dog", "white", 5000);
+            List<Pet> pets = new List<Pet>() { pet1, pet2, pet3 };
+            string request = JsonConvert.SerializeObject(pets);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+            //when
+            await client.PostAsync("petStore/addNewPets", requestBody);
+            var response = await client.GetAsync("PetStore/WangCai");
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            Pet actualPet = JsonConvert.DeserializeObject<Pet>(responseString);
+            Assert.Equal(pet3, actualPet);
+        }
+
+        [Fact]
+        public async void Should_remove_pet_after_sale()
+        {
+            //given
+            TestServer server = new TestServer(new WebHostBuilder().UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("petStore/removePets");
+            Pet pet1 = new Pet("HuanHuan", "dog", "white", 5000);
+            Pet pet2 = new Pet("DaHuang", "dog", "white", 5000);
+            Pet pet3 = new Pet("WangCai", "dog", "white", 5000);
+            List<Pet> pets = new List<Pet>() { pet1, pet2, pet3 };
+            string request = JsonConvert.SerializeObject(pets);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+            //when
+            await client.PostAsync("petStore/addNewPets", requestBody);
+            var response = await client.DeleteAsync("petStore/RemoveSoldPet/WangCai");
+            //then
```

Comment:
you need to check get method for the removed pet to see whether it's removed or not. the code you implemented here doesn't help you find the bug in your api controller method.